tools/src/test/scala/com/nvidia/spark/rapids/tool/profiling/AutoTunerSuite.scala

```diff
@@ -0,0 +1,163 @@
+
+  test("test system property file - exists") {
+    val systemPropsFilePath = s"$systemPropsDir/system_props.yaml"
+    val systemInfo = AutoTuner.parseSystemInfo(systemPropsFilePath)
+    assert(systemInfo != null)
+  }
+
+  test("test system property file - does not exists") {
+    val systemPropsFilePath = s"$systemPropsDir/system_props_fail.yaml"
+    val systemInfo = AutoTuner.parseSystemInfo(systemPropsFilePath)
+    assert(systemInfo == null)
+  }
+
+  test("test system property file - without gpu") {
+    val systemPropsFilePath = s"$systemPropsDir/system_props_without_gpu.yaml"
+    val systemInfo = AutoTuner.parseSystemInfo(systemPropsFilePath)
+    assert(systemInfo != null)
+    assert(systemInfo.gpu_props == null)
+  }
+
+  test("test system property file - without num-workers") {
+    val systemPropsFilePath = s"$systemPropsDir/system_props_without_num_workers.yaml"
+    val systemInfo = AutoTuner.parseSystemInfo(systemPropsFilePath)
+    assert(systemInfo != null)
+    assert(systemInfo.num_workers.isEmpty)
+  }
+
+  test("test convert from human readable size") {
+    val rand = new Random()
+    AutoTuner.SUPPORTED_UNITS.zipWithIndex.foreach {
+      case (size, index) =>
+        val randomValue = "%.3f".format(10 * rand.nextDouble())
+        val testSize = s"$randomValue$size"
+        val expectedSize = (randomValue.toDouble * Math.pow(1024, index)).toLong
+        val resultSize = AutoTuner.convertFromHumanReadableSize(testSize)
+        assert(resultSize == expectedSize)
+    }
+  }
+
+  test("test convert to human readable size") {
+    val rand = new Random()
+    AutoTuner.SUPPORTED_UNITS.zipWithIndex.foreach {
+      case (size, index) =>
+      val randomValue = 1 + rand.nextInt(9)
+      val testSize = (randomValue * Math.pow(1024, index)).toLong
+      val expectedSize = s"$randomValue$size"
+      val resultSize = AutoTuner.convertToHumanReadableSize(testSize)
+      assert(resultSize == expectedSize)
+    }
+  }
+
+  test("test compare spark version") {
+    assert(AutoTuner.compareSparkVersion("3", "1") > 0)
+    assert(AutoTuner.compareSparkVersion("3.3.1", "3.0.0") > 0)
+    assert(AutoTuner.compareSparkVersion("4.1.5", "4.15") < 0)
+    assert(AutoTuner.compareSparkVersion("2.0", "2.0.0") == 0)
+    assert(AutoTuner.compareSparkVersion("2", "2.0.0") == 0)
+  }
+
+  private def testSingleEvent(
+    eventLog: String,
+    systemPropsFilePath: String,
+    expectedFilePath: String): Unit = {
+
+    TrampolineUtil.withTempDir { tempDir =>
+      val appArgs = new ProfileArgs(Array(
+        "--auto-tuner",
+        "--worker-info",
+        systemPropsFilePath,
+        "--output-directory",
+        tempDir.getAbsolutePath,
+        eventLog))
+      val (exit, _) = ProfileMain.mainInternal(appArgs)
+      assert(exit == 0)
+      val resultsFile = fromFile(tempDir + s"/${Profiler.SUBDIR}/local-1655514789396/profile.log")
```

Comment:
how is the application name (local-1655514789396) alwasy the same here?

Reply:
Since, I am using the same event-log for all UTs. I will update it to a variable.